src/integrationTest/java/org/opensearch/security/api/CertificatesRestApiIntegrationTest.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+package org.opensearch.security.api;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.StringJoiner;
+import java.util.stream.Collectors;
+
+import com.carrotsearch.randomizedtesting.RandomizedContext;
+import com.google.common.collect.ImmutableList;
+import com.fasterxml.jackson.databind.JsonNode;
+import org.apache.commons.lang3.tuple.Pair;
+import org.junit.Test;
+
+import org.opensearch.security.dlic.rest.api.Endpoint;
+import org.opensearch.security.dlic.rest.api.ssl.CertificateType;
+import org.opensearch.test.framework.certificate.TestCertificates;
+import org.opensearch.test.framework.cluster.LocalOpenSearchCluster;
+import org.opensearch.test.framework.cluster.TestRestClient;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.opensearch.security.OpenSearchSecurityPlugin.PLUGINS_PREFIX;
+import static org.opensearch.security.dlic.rest.api.RestApiAdminPrivilegesEvaluator.CERTS_INFO_ACTION;
+import static org.opensearch.security.support.ConfigConstants.SECURITY_RESTAPI_ADMIN_ENABLED;
+
+public class CertificatesRestApiIntegrationTest extends AbstractApiIntegrationTest {
```

Comment:
Does this test class configure `plugins.security.ssl.client.external_context_id` anywhere? I'm trying to read up more about this setting from the docs, but still don't quite understand the use of it. The [OpenSearch docs](https://opensearch.org/docs/latest/install-and-configure/configuring-opensearch/security-settings/#transport-layer-security-settings) mention:

> Provides the transport client an ID to use for an external SSL context.

Can you share an example config that uses this setting?

Reply:
Gotcha. I was wondering if the checks with `!ExternalSecurityKeyStore.hasExternalSslContext(settings)` were required. 

The Transport client was removed in 2.0.0, so judging by the documentation it looks like that setting cannot actually be used.

Reply:
ahh ok i will remove it